File: examples/validate.py
import base64
import hmac
import hashlib
import json

ass = """eyJsaWNlbnNlIjp7ImlkIjoidGVzdCIsIm1ldGEiOnt9LCJ2YWxpZF9mcm9tIjoiMjAwMC0xLTEiLCJ2YWxpZF91bnRpbCI6IjMwMDAtMS0xIn0sInNpZ25hdHVyZSI6ImVhYzJkMjI2ZjA0NTFjMmQ5NTM2NzkxZDg2NDEyMjRhZWFmMjkwY2NmZjEzYWQxZDE0YmYxY2U2OGMyYzJmMmQifQ== """

d = json.loads(base64.standard_b64decode(ass))

parts = []
parts += [d['license']['valid_from']]
parts += [d['license']['valid_until']]
parts += [d['license']['id']]
for k, v in d['license']['meta']:
    parts += [v]
to_hash = "\n".join(parts)

digest_maker = hmac.new(bytes('SECRET', encoding='utf8'), None, hashlib.sha256)
digest_maker.update(bytes(to_hash, encoding='utf8'))

print(digest_maker.hexdigest() == d['signature'])

File: examples/validate.js
const crypto = require('crypto')

const ass = `eyJsaWNlbnNlIjp7ImlkIjoidGVzdCIsIm1ldGEiOnt9LCJ2YWxpZF9mcm9tIjoiMjAwMC0xLTEiLCJ2YWxpZF91bnRpbCI6IjMwMDAtMS0xIn0sInNpZ25hdHVyZSI6ImVhYzJkMjI2ZjA0NTFjMmQ5NTM2NzkxZDg2NDEyMjRhZWFmMjkwY2NmZjEzYWQxZDE0YmYxY2U2OGMyYzJmMmQifQ==`

const d = JSON.parse((Buffer.from(ass, 'base64').toString('ascii')));
const parts = [];
parts.push(d['license']['valid_from']);
parts.push(d['license']['valid_until']);
parts.push(d['license']['id']);
const keys = Object.keys(d['license']['meta']).map(k => k);
keys.sort();
keys.forEach(v => parts.push(d['license']['meta'][v]));

const to_hash = parts.join("\n");

const result = crypto.createHmac('sha256', 'SECRET')
    .update(to_hash)
    .digest('hex')

console.log(result === d['signature'])
